pkg/registry/extensions/daemonset/strategy.go

```diff
@@ -70,6 +70,15 @@ func (daemonSetStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, ol
 	// update is not allowed to set TemplateGeneration
 	newDaemonSet.Spec.TemplateGeneration = oldDaemonSet.Spec.TemplateGeneration
 
+	// Update is not allowed to set Spec.Selector for all groups/versions except extensions/v1beta1.
+	// If RequestInfo is nil, it is better to revert to old behavior (i.e. allow update to set Spec.Selector)
+	// to prevent unintentionally breaking users who may rely on the old behavior.
+	// TODO(#50791): after v1beta1 is retired, move selector immutability check to validation codes
+	requestInfo, found := genericapirequest.RequestInfoFrom(ctx)
+	if found && !(requestInfo.APIGroup == "extensions" && requestInfo.APIVersion == "v1beta1") {
```

Comment:
this might be clearer as a switch of the versions we expect:

```
if requestInfo, found := genericapirequest.RequestInfoFrom(ctx); found {
  groupVersion := schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}
  switch groupVersion {
    case extensionsv1beta1.SchemeGroupVersion:
      // no-op for compatibility
    case appsv1beta2.SchemeGroupVersion:
      // disallow mutation of selector
      newDaemonSet.Spec.Selector = oldDaemonSet.Spec.Selector
    default:
      // error? panic? default to disallowing mutation?
  }
}
```